include/circt/Dialect/RTGTest/IR/RTGTestAttributes.td

```diff
@@ -0,0 +1,60 @@
+//===- RTGTestAttributes.td - RTGTest attributes -----------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This describes the RTGTest attributes.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_RTGTEST_IR_RTGTESTATTRIBUTES_TD
+#define CIRCT_DIALECT_RTGTEST_IR_RTGTESTATTRIBUTES_TD
+
+include "mlir/IR/AttrTypeBase.td"
+include "mlir/IR/EnumAttr.td"
+
+// Flat allocation of unique IDs to all registers. The actual ID value does not
+// matter.
+def RegisterAttr : I32EnumAttr<
```

Comment:
I think ideally this attribute should be of IntegerRegisterType, not I32. It's not a big deal for now though.